### DISCUSSION TITLE
Q.Cells identical EVC as Solax

### DISCUSSION THREAD

**Discussion by moerlien** [PatrikTrestik/homeassistant-solax-http#31] [Q&A] [answered]
Hi, I have a Q.Cells Q.Home edrive EVC which should be identical to the Solax EVC.

I was trying to get HTTP API working in HA, but I don't get any entities.

If I change the admin password of the webinterface, does the API password changes too or does it stay the SN?

Is the IP of the HTTP API Integration the one of the Wallbox in my WiFi?

Does the HTTP API Integration works with only the EVC running or has the Inverter to be online too? And do I then habe to use the Inverter IP?

Thanks for your Support.

**Comment by PatrikTrestik** [comment]
> If I change the admin password of the webinterface, does the API password changes too or does it stay the SN?

It is same password. Default is not SN, but Registration number (SolaX has little chaos in names).

> Is the IP of the HTTP API Integration the one of the Wallbox in my WiFi?

Yes, same address.

> Does the HTTP API Integration works with only the EVC running or has the Inverter to be online too? And do I then habe to use the Inverter IP?

It is not required to have Inverter up. EVC is completely independent.


Are you sure you have SolaX firmware? Nothin was tested with other EVC and I have no information if your charger supports API.

**Comment by moerlien** [comment]
Thanks for your Feedback.If I want to Start from scratch, what do I need to explore the HTTP API? I would like to check what differences are between solax and q.cells firmware. Perhaps it's just another port on which the communication works? Or your requests in the HA integration have a different naming than q.cells registers?Or should I flash it with solax firmware and hope that it will work with my Inverter, when it goes online? 😄

**Reply by PatrikTrestik** [reply]
If you want to play and you know how, try this:

Send POST request to http://<EVC ip address>/
with payload in plaintext: optType=ReadRealTimeData&pwd=<put your password here>

You should receive json response back. If you do send it to me and I can check if mapping is similar.

**Reply by moerlien** [reply]
I tried to get a Response via curl, but it seems not to work. Any idea? Thought it could be the password, but it makes no difference if I use the SN or the Reg #

curl -d "optType=ReadRealTimeData&pwd=" -X POST http://192.168.178.175

The response is everytime 0:code,message:"failed"}

the ip of the EVC is reachable via ping.

**Reply by moerlien** [reply]
Okay, it seems there is also a way to get this response over solaxcloud.com with http API and tokenID. But it doesn't work either. it gets result "no auth!" with Reg#, SN or WebUI Pwd and "sn invalid!" with no pwd.

So it needs either another password or it does not work with qcells firmware on evc.

**Reply by moerlien** [reply]
Will try another run with WiFi of the EVC. I've read anywhere that it could work.

**Reply by PatrikTrestik** [reply]
This reply shows that API is there. It is good news.

Few ideas:
- curl is not sending plain text body. Try with Insomnia or Postman to have more control. Or just play with HA integration and different passwords.
- API is different. Which android app you use to connect EVC? Is there option for local connect? Try. I can decompile app and look what API calls it uses.

**Reply by moerlien** [reply]
I use the qhome App and the solax App as well. Also in the Solax Cloud the EVC is connected and can be read or configurated.

Will try Insomnia or postman later. 

HA Integration says

Logger: custom_components.solax_http.plugin_solax_ev_charger
Quelle: custom_components/solax_http/plugin_solax_ev_charger
unrecognized inverter type - serial number : unknown

or

Logger: custom_components.solax_http.plugin_solax_ev_charger
Quelle: custom_components/solax_http/plugin_solax_ev_charger
Attempt to read serialnumber failed

or

Logger: custom_components.solax_http.coordinator
Quelle: custom_components/solax_http/coordinator.py:120
Failed to read data from http: 0:code,message:"failed"}

**Reply by PatrikTrestik** [reply]
First reply is good.
It says that it does not know your EVC type, but API was probably successful.

Send me SN (not RN or passwd 😁). I'll add it to integration. 

Also output of APi will be good for me to compare with solax.

**Reply by moerlien** [reply]
Can I send you the SN in private?

**Reply by PatrikTrestik** [reply]
SN is ok. Reg number is secret.

**Reply by moerlien** [reply]
You got it via Mail as well, right?

**Reply by PatrikTrestik** [reply]
I need this info:
![image](https://github.com/PatrikTrestik/homeassistant-solax-http/assets/17616747/6b8b927f-20f5-4a30-b88f-1cc540052eb4)

**Reply by moerlien** [reply]
Do you need anything else from me? Can I do something from my side?

**Reply by PatrikTrestik** [reply]
Hi. Actually, I need time :)
There was busy work time for me so I was only able to check that your version is Ok and supported. I have to dive deep and check.

I'll prepare and test curl command which works for me and than you should check.

**Reply by moerlien** [reply]
Please apologise. I didn't mean to rush you. If there is anything I can do to help, please let me know.

**Reply by PatrikTrestik** [reply]
No need to apologize. This is my hobby and I like to do my hobby 😄 Unfortunately work has higher priority.

I checked code and tested assess. Here are few thinks you should try:
1) Login to web UI and note password used for access. I suggest to change password to something really simple during tests (for example "TEST123") and change it back after testing.
2) Try this command: curl -d "optType=ReadRealTimeData&pwd=TEST123" http://EVC.IP/
Replace TEST123 >> your actual password. EVC.IP >> ip of your EVC as seen in web ui.

Let me know results.

**Reply by PatrikTrestik** [reply]
@moerlien any progress?

**Reply by moerlien** [reply]
No, unfortunately not. Got a firmware Upgrade. Now it seems there is no Web UI on the EVC IP anymore. Tried to find open ports, but that also has not worked. 

In Former Firmware i wasn't able to login to Web ui. So the curl command did also not work.

I will try to connect the evc via modbus Adapter. But hadn't time to get it running yet. 

Anything Else I could do?

**Reply by PatrikTrestik** [reply]
Scan all open ports on EVC IP. Nmap tool can help - use full scan.

If there is some port open we can look what it is.

Is there Android app which can communicate locally with EVC? Reverse engineering of the app caould help.

**Reply by moerlien** [reply]
I have access via QCells as well as Solax Android App. The access is via WiFi, because I added Modbus cable connection afterwards.

**Reply by moerlien** [reply]
I have identified 
PORT     STATE SERVICE
5353/udp open  zeroconf

on my EVC. Does that help?
The Webinterface is also back on Port 80, but I don't know the login credentials.

**Reply by moerlien** [reply]
I saw, that I can get an backup password via Solax Cloud System. I tried your curl command again on EVC IP with this backup password and I think I got an usefull answer :)

{"SN":"REGISTERINGNUMBER","ver":"3.006.01","type":1,"Data":[0,0,22981,22903,22855,0,0,0,0,0,0,0,0,0,4167,0,64,35463,65488,65525,65514,65526,65475,0,31,0,0,14,0,0,0,0,0,5006,4998,5002,14651,6159,6153,2,0,0,0,0,0,0,0,0,0,100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,10803,5645,6153,0,0,1,1,0,0,120,1196,1000,120,49,0,0,1,1,0],"Information":[11.000,1,"SERIALNUMBER",1,1.14,1.01,0.00,0.00,0.00,1],"OCPPServer":"","OCPPChargerId":"","QRContent":""}

Does that help you?

**Reply by PatrikTrestik** [reply]
It looks good. Response is nearly exact as one of SolasX EVC.
Integration should work with it.

Points to be solved:
1) I expect that you edited SERIALNUMBER for security reasons, but response contains correct one.
Integration uses it to identify type of EVC. Please send me first part (like C31103) so I can check it.

2) I hope that positions in Data field are same as for SolaX. If not we have to investigate mapping.
You will see onece integration starts.

3) Try to log to web UI. User is admin and password is same as you used for curl.

4) Test integration. If your SN looks like my example, it should work.

**Reply by moerlien** [reply]
Hi Patrik,
1. right, first Part is C31101
2. HA Integration finds no entities
3. Login works, it is the same Web ui as if i use Pocket wifi of evc with registration number as password
4. where do i enter the SN? I have only fields for IP and API pasword

**Reply by PatrikTrestik** [reply]
I see no problem for integration to work.

SN is read from API automaticaly. Check if you have correct IP and password (same as you use for Web UI).

Thank look to HA logs and check if any error message.

**Reply by moerlien** [reply, marked as answer]
It is working for a while now. Thanks for your amazing Support.

Do you know if I could see and track the use of the RFID Cards when charging?

**Reply by PatrikTrestik** [reply]
I think that RFID could be tracked, but it will require some research.
I have it on my list when I get home and will have access to hardware. 

Please open new Issue specific for RFID so I can track it.
